api/server/fntracer.go

```diff
@@ -0,0 +1,130 @@
+package server
+
+import (
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/log"
+	"strings"
+)
+
+// FnTracer is a custom Tracer which wraps another another tracer
+// its main purpose is to wrap the underlying Span in a FnSpan,
+// which adds some extra behaviour required for sending tracing spans to prometheus
+type FnTracer struct {
+	wrappedTracer opentracing.Tracer
```

Comment:
one cool trick w/ go is that you can embed interfaces / structs and inherit their methods. for this implementation of an `opentracing.Tracer` we only need to override `StartSpan`, so if the `wrappedTracer` were embedded then we wouldn't need to have pass through methods for `Inject` or `Extract` since the embedded `opentracing.Tracer` would be seamlessly used. the embedded tracer can be addressed by doing `thisFnTracer.Tracer`.

Reply:
Yes, that's very cool. Done.